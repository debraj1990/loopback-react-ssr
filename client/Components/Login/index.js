import React, { Component } from 'react';
import Form from '../Form';



class Login extends Component{
    
        state = {
            email:'',
            password:'',
        }

    onChange = ({target}) => {
        this.setState({[target.name]:target.value})
    }

    render(){
        const {email, password} = this.state
        return (
            <div>
                <Form
                  email={email}
                  password={password}
                  onChange={this.onChange}
                  type='Login'
                />
            </div>
        )
    }
}



export default Login