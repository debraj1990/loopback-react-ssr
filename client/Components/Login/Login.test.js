import React from 'react';
import {configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import { createMemoryHistory } from 'history'

import Login from './'
import { c } from 'strong-globalize/lib/globalize';


configure({adapter: new Adapter });


describe('Login Component', () => {
  
  it('component should render the compoent without crushing', () => {
    const history = createMemoryHistory('/login')
    const component = shallow(<Login history={history} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  });

  it('should check whether the localstorage id is saved', () => {

    localStorage.setItem('__id__', 'dkdkskkekekdkkdkdkdkk')
    expect(localStorage.getItem('__id__')).toBe('dkdkskkekekdkkdkdkdkk')
  })

    it('should render the forms', () => {
        const wrapper = shallow(<Login error={false} />)
        expect(wrapper.find('Form')).toHaveLength(1)
      });

    
})