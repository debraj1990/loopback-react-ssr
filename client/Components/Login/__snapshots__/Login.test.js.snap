// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component component should render the compoent without crushing 1`] = `
<div>
  <h3
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    Hi Welcome Login 
  </h3>
  <Form
    email=""
    onChange={[Function]}
    onSubmit={[Function]}
    password=""
    type="Login"
  />
</div>
`;
